src/createClient.js

```diff
@@ -34,10 +34,19 @@ function createClient (options) {
   const client = new Client(false, version.minecraftVersion, options.customPackets, hideErrors)
 
   tcpDns(client, options)
-  if (options.auth === 'microsoft') {
-    microsoftAuth.authenticate(client, options)
-  } else {
-    auth(client, options)
+  switch (options.auth) {
+    case 'mojang':
+      console.warn('[deprecated] mojang auth servers no longer accept mojang accounts to login. convert your account.\nhttps://help.minecraft.net/hc/en-us/articles/4403181904525-How-to-Migrate-Your-Mojang-Account-to-a-Microsoft-Account')
+      auth(client, options)
+      break
+    case 'microsoft':
+      microsoftAuth.authenticate(client, options)
+      break
+    case 'offline':
+    default:
+      client.username = options.username
```

Comment:
why that change?

Reply:
> why that change?

We need to set a username for the login_start packet in setProtocol.